src/name.rs

```diff
@@ -617,6 +617,62 @@ impl NamespaceResolver {
             None => ResolveResult::Unbound,
         }
     }
+
+    #[inline]
+    pub fn iter(&self) -> PrefixIter {
+        PrefixIter {
+            resolver: self,
+            bindings_cursor: 2, // We initialize the cursor to 2 to skip the two default namespaces xml: and xmlns:
+        }
+    }
+}
+
+/// Iterator on the current declared prefixes.
+///
+/// See [`NsReader::prefixes`](crate::NsReader::prefixes) for documentation.
+pub struct PrefixIter<'a> {
+    resolver: &'a NamespaceResolver,
+    bindings_cursor: usize,
+}
+
+impl<'a> Iterator for PrefixIter<'a> {
+    type Item = (PrefixDeclaration<'a>, Namespace<'a>);
+
+    fn next(&mut self) -> Option<(PrefixDeclaration<'a>, Namespace<'a>)> {
+        while let Some(namespace_entry) = self.resolver.bindings.get(self.bindings_cursor) {
+            self.bindings_cursor += 1; // We increment for next read
+
+            // We check if the key has not been overriden by having a look at the namespaces declared after in the array
+            if self.resolver.bindings[self.bindings_cursor..]
+                .iter()
+                .any(|ne| {
+                    namespace_entry.prefix(&self.resolver.buffer)
+                        == ne.prefix(&self.resolver.buffer)
+                })
+            {
+                continue; // Overriden
+            }
```

Comment:
It seems we can return `None` here because on all next iterations we also enter in this `if`, don't we?

Also, `namespace_entry.prefix(&self.resolver.buffer)` could be calculated outside of closure and I have a feeling that the whole this loop can be embedded.

Reply:
> Also, namespace_entry.prefix(&self.resolver.buffer) could be calculated outside of closure and I have a feeling that the whole this loop can be embedded.

Indeed. Done. Thank you!

> It seems we can return None here because on all next iterations we also enter in this if, don't we?

I don't think so because we check only if a namespace of the same name is declared, not if an other namespace in general is declared.

Let's say we have
```xml
<foo:outer xmlns:foo="a" xmlns:bar="b">
  <foo:inner xmlns:foo="c"></foo:inner>
</foo:outer>
```
The `bindings` list when parsing `foo:inner` is going to be something like `[("foo", "a"), ("bar", "b"), ("foo", "c")]` so, when we iterate on it we don't want to stop iterating on the first `("foo", "a")` because it has been overriden by `("foo", "c")`.